diff -ruN a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
--- a/drivers/gpu/drm/i915/intel_dp.c	2018-02-14 18:32:10.214086906 -0500
+++ b/drivers/gpu/drm/i915/intel_dp.c	2018-02-14 19:44:43.304779276 -0500
@@ -114,7 +114,7 @@
 {
 	struct intel_digital_port *intel_dig_port = dp_to_dig_port(intel_dp);
 
-	return intel_dig_port->base.type == INTEL_OUTPUT_EDP;
+    return intel_dig_port->base.type == INTEL_OUTPUT_EDP || strncmp(intel_dp->attached_connector->base.name, "DP-3", 4) == 0;
 }
 
 static struct drm_device *intel_dp_to_dev(struct intel_dp *intel_dp)
@@ -5150,7 +5150,7 @@
 	if (INTEL_GEN(dev_priv) < 5)
 		return false;
 
-	if (INTEL_GEN(dev_priv) < 9 && port == PORT_A)
+	if (INTEL_GEN(dev_priv) < 9 && port == PORT_D)
 		return true;
 
 	return intel_bios_is_port_edp(dev_priv, port);
diff -ruN a/drivers/gpu/drm/i915/intel_lvds.c b/drivers/gpu/drm/i915/intel_lvds.c
--- a/drivers/gpu/drm/i915/intel_lvds.c	2018-02-14 18:32:10.237078040 -0500
+++ b/drivers/gpu/drm/i915/intel_lvds.c	2018-02-14 20:00:25.314830346 -0500
@@ -620,10 +620,10 @@
 static const struct dmi_system_id intel_no_lvds[] = {
 	{
 		.callback = intel_no_lvds_dmi_callback,
-		.ident = "Apple Mac Mini (Core series)",
+		.ident = "modded Thinkpad X320",
 		.matches = {
-			DMI_MATCH(DMI_SYS_VENDOR, "Apple"),
-			DMI_MATCH(DMI_PRODUCT_NAME, "Macmini1,1"),
+			DMI_MATCH(DMI_SYS_VENDOR, "LENOVO"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "4290FP2"),
 		},
 	},
 	{
 		.callback = intel_no_lvds_dmi_callback,
-		.ident = "Apple Mac Mini (Core 2 series)",
+		.ident = "modded Thinkpad X330",
 		.matches = {
-			DMI_MATCH(DMI_SYS_VENDOR, "Apple"),
-			DMI_MATCH(DMI_PRODUCT_NAME, "Macmini2,1"),
+			DMI_MATCH(DMI_SYS_VENDOR, "LENOVO"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "2324B14"),
 		},
 	},
 	{
